File: projects/payme/java/domain/repository/CardRepository.java
package jk.uz.pdp.domain.repository;

import jk.uz.pdp.domain.exception.DataNotFoundException;
import jk.uz.pdp.domain.exception.EPAMPaymentServerException;
import jk.uz.pdp.domain.model.Card;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public abstract class CardRepository implements BaseCrudRepository<Card, Long> {
    private final List<Card> cardList = new ArrayList<>();
    private Long id = 0L;

    @Override
    public void create(Card card) {
        if(isExist(card.getNumber())) {
            throw new EPAMPaymentServerException("Card already exists");
        }
        card.setId(id++);
        cardList.add(card);
    }

    @Override
    public Card getById(Long id) {
        for (Card card:cardList) {
            if (card.getId().equals(id)){
                return card;
            }
        }
        throw new DataNotFoundException("There is not card with id: " + id);
    }

    @Override
    public List<Card> getList() {
        return cardList;
    }

    @Override
    public Card update(Card update) {
        int i = 0;
        for (Card card:cardList) {
            if (card.getId().equals(update.getId())){
                cardList.set(i,update);
                return card;
            }
            i++;
        }
        throw new DataNotFoundException("There is not card with this id " + update.getId());
    }

    @Override
    public void delete(Long id) {
        for (Card card:cardList) {
            if (card.getId().equals(id)){
                cardList.remove(card);
                break;
            }
        }
        throw new DataNotFoundException("There is not card with id: " + id);
    }

    public List<Card> getCardsByOwnerId(UUID ownerId) {
        List<Card> cards = new ArrayList<>();
        for(Card card:cardList) {
            if(card.getOwnerId().equals(ownerId)) {
                cards.add(card);
            }
        }

        if(cards.isEmpty()) {
            throw new DataNotFoundException("This user doesn't have any cards");
        }
        return cards;
    }

    private boolean isExist(String cardNumber) {
        for(Card card:cardList) {
            if(card.getNumber().equals(cardNumber)) {
                return true;
            }
        }
        return false;
    }
}

File: projects/payme/src/repository/card-repository.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardRepository = void 0;
var CardRepository = /** @class */ (function () {
    function CardRepository() {
        this.list = [];
        this.counter = 0;
    }
    CardRepository.prototype.create = function () {
        var cards = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            cards[_i] = arguments[_i];
        }
        for (var _a = 0, cards_1 = cards; _a < cards_1.length; _a++) {
            var card = cards_1[_a];
            if (this.isExist(card.number)) {
                throw new Error("card(".concat(card.number, ") already exists"));
            }
            card.setId(++this.counter);
            this.list.push(card);
        }
    };
    CardRepository.prototype.isExist = function (cardNumber) {
        for (var _i = 0, _a = this.list; _i < _a.length; _i++) {
            var card = _a[_i];
            if (card.number === cardNumber)
                return true;
        }
        return false;
    };
    CardRepository.prototype.getById = function (cardID) {
        for (var _i = 0, _a = this.list; _i < _a.length; _i++) {
            var card = _a[_i];
            if (card.getId() === cardID) {
                return card;
            }
        }
        throw new Error("card not found");
    };
    CardRepository.prototype.getList = function () {
        return this.list;
    };
    CardRepository.prototype.getCardsByOwnerId = function (ownerID) {
        var cards = [];
        for (var _i = 0, _a = this.list; _i < _a.length; _i++) {
            var card = _a[_i];
            if (card.ownerId === ownerID)
                cards.push(card);
        }
        if (cards.length === 0)
            throw new Error("No cards found");
        return cards;
    };
    return CardRepository;
}());
exports.CardRepository = CardRepository;
